# origin: Debian (herbert, many bugs)
# cset: n/a
# inclusion: superseded by 076-078
# description: misc. isofs fixes
# revision date: 2004-09-05

diff -urN kernel-source-2.4.26/fs/isofs/dir.c kernel-source-2.4.26-1/fs/isofs/dir.c
--- kernel-source-2.4.26/fs/isofs/dir.c	2002-02-26 06:38:08.000000000 +1100
+++ kernel-source-2.4.26-1/fs/isofs/dir.c	2002-04-13 21:45:40.000000000 +1000
@@ -63,7 +63,8 @@
 			break;
 
 		/* Convert remaining ';' to '.' */
-		if (c == ';')
+		/* Also '/' to '.' (broken Acorn-generated ISO9660 images) */
+		if (c == ';' || c == '/')
 			c = '.';
 
 		new[i] = c;
diff -urN kernel-source-2.4.26/fs/isofs/inode.c kernel-source-2.4.26-1/fs/isofs/inode.c
--- kernel-source-2.4.26/fs/isofs/inode.c	2002-11-29 10:53:15.000000000 +1100
+++ kernel-source-2.4.26-1/fs/isofs/inode.c	2002-11-30 15:49:27.000000000 +1100
@@ -1148,7 +1148,10 @@
 	struct iso_directory_record * tmpde = NULL;
 	unsigned int de_len;
 	unsigned long offset;
-	int volume_seq_no, i;
+	int i;
+#ifndef IGNORE_WRONG_MULTI_VOLUME_SPECS
+	int volume_seq_no;
+#endif
 
 	bh = sb_bread(inode->i_sb, block);
 	if (!bh)
@@ -1190,14 +1193,6 @@
 		inode->i_mode = inode->i_sb->u.isofs_sb.s_mode;
 		inode->i_nlink = 1;
 	        inode->i_mode |= S_IFREG;
-		/* If there are no periods in the name,
-		 * then set the execute permission bit
-		 */
-		for(i=0; i< de->name_len[0]; i++)
-			if(de->name[i]=='.' || de->name[i]==';')
-				break;
-		if(i == de->name_len[0] || de->name[i] == ';')
-			inode->i_mode |= S_IXUGO; /* execute permission */
 	}
 	inode->i_uid = inode->i_sb->u.isofs_sb.s_uid;
 	inode->i_gid = inode->i_sb->u.isofs_sb.s_gid;
@@ -1280,9 +1275,6 @@
 		test_and_set_gid(&inode->i_gid, inode->i_sb->u.isofs_sb.s_gid);
 	}
 
-	/* get the volume sequence number */
-	volume_seq_no = isonum_723 (de->volume_sequence_number) ;
-
     /*
      * Multi volume means tagging a group of CDs with info in their headers.
      * All CDs of a group must share the same vol set name and vol set size
@@ -1292,6 +1284,9 @@
      * of ftp://ftp.ecma.ch/ecma-st/Ecma-119.pdf (ECMA 119 2nd Ed = ISO 9660)
      */
 #ifndef IGNORE_WRONG_MULTI_VOLUME_SPECS
+	/* get the volume sequence number */
+	volume_seq_no = isonum_723 (de->volume_sequence_number) ;
+
 	/*
 	 * Disable checking if we see any volume number other than 0 or 1.
 	 * We could use the cruft option, but that has multiple purposes, one

